GiniSwitchSDK/GiniSwitchSDK/Classes/ExtractionCollection.swift

```diff
@@ -8,34 +8,92 @@
+    public var customerAddress:Extraction<ContractAddressValue>?
+    public var energyMeterNumber:Extraction<String>?
+    public var consumption:Extraction<AmountValue>?
+    public var consumptionDuration:Extraction<AmountValue>?
+    public var billingAmount:Extraction<AmountValue>?
+    public var paidAmount:Extraction<AmountValue>?
+    public var amountToPay:Extraction<AmountValue>?
+    public var documentDate:Extraction<String>?
+    public let links:ResponseLinks?
+    
+    private enum CodingKeys : String, CodingKey {
+        case companyName = "companyName"
+        case customerAddress = "customerAddress"
+        case energyMeterNumber = "energyMeterNumber"
+        case consumption = "consumption"
+        case consumptionDuration = "consumptionDuration"
+        case billingAmount = "billingAmount"
+        case paidAmount = "paidAmount"
+        case amountToPay = "amountToPay"
+        case documentDate = "documentDate"
+        case links = "_links"
+    }
     
-    public init() {
-        extractions = []
-        super.init(href: "")
+    private enum ValueKeys : String, CodingKey {
+        case value = "value"
+        case alternatives = "alternatives"
     }
     
-    public init(collection:[Extraction]) {
-        extractions = collection
-        super.init(href: "")
+    public required init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        links = try container.decode(ResponseLinks.self, forKey: .links)
+        companyName = ExtractionCollection.parse(type: String.self, decoder: decoder, key: .companyName, name: companyNameKey, container: container)
+        customerAddress = ExtractionCollection.parse(type: ContractAddressValue.self, decoder: decoder, key: .customerAddress, name: customerAddressKey, container: container)
+        energyMeterNumber = ExtractionCollection.parse(type: String.self, decoder: decoder, key: .energyMeterNumber, name: energyMeterNumberKey, container: container)
+        consumption = ExtractionCollection.parse(type: AmountValue.self, decoder: decoder, key: .consumption, name: consumptionKey, container: container)
+        consumptionDuration = ExtractionCollection.parse(type: AmountValue.self, decoder: decoder, key: .consumptionDuration, name: consumptionDurationKey, container: container)
+        billingAmount = ExtractionCollection.parse(type: AmountValue.self, decoder: decoder, key: .billingAmount, name: billingAmountKey, container: container)
+        paidAmount = ExtractionCollection.parse(type: AmountValue.self, decoder: decoder, key: .paidAmount, name: paidAmountKey, container: container)
+        amountToPay = ExtractionCollection.parse(type: AmountValue.self, decoder: decoder, key: .amountToPay, name: amountToPayKey, container: container)
+        documentDate = ExtractionCollection.parse(type: String.self, decoder: decoder, key: .documentDate, name: documentDateKey, container: container)
```

Comment:
Me too, but I don't see how. Normally Codable handles everything, but since extractions are actually a generic type, it cannot do it automatically. At least I didn't find a way.